code/modules/surgery/bodyparts/bodypart_effects.dm

```diff
@@ -0,0 +1,136 @@
+	var/is_active = FALSE
+
+/datum/status_effect/bodypart_effect/on_creation(mob/living/new_owner, obj/item/bodypart/bodypart)
+	. = ..()
+
+	add_bodypart(bodypart)
+
+/// Merge a bodypart into the effect
+/datum/status_effect/bodypart_effect/proc/add_bodypart(bodypart)
+	RegisterSignal(bodypart, COMSIG_BODYPART_REMOVED, PROC_REF(on_bodypart_removed))
+	RegisterSignal(bodypart, COMSIG_QDELETING, PROC_REF(on_bodypart_destroyed))
+
+	bodyparts.Add(bodypart)
+
+	if(!is_active && bodyparts.len >= minimum_bodyparts)
+		activate()
+
+/// Remove a bodypart from the effect. Deleting = TRUE is used during clean-up phase
+/datum/status_effect/bodypart_effect/proc/remove_bodypart(obj/item/bodypart/bodypart, deleting = FALSE)
+	UnregisterSignal(bodypart, COMSIG_BODYPART_REMOVED)
+
+	bodyparts.Remove(bodypart)
+
+	if(deleting)
+		return
+
+	if(bodyparts.len == 0)
+		qdel(src)
+
+	else if(is_active && bodyparts.len < minimum_bodyparts)
+		deactivate()
+
+/// Signal called when a bodypart is removed
+/datum/status_effect/bodypart_effect/proc/on_bodypart_removed(obj/item/bodypart/bodypart)
+	SIGNAL_HANDLER
+
+	remove_bodypart(owner, bodypart)
+
+/// Signal called when a bodypart is destroyed. Destruction of a bodypart doesn't necessarily drop it
+/datum/status_effect/bodypart_effect/proc/on_bodypart_destroyed(obj/item/bodypart/bodypart)
+	SIGNAL_HANDLER
+
+	remove_bodypart(bodypart)
+
+/// Activate some sort of effect when a threshold is reached
+/datum/status_effect/bodypart_effect/proc/activate()
+	PROTECTED_PROC(TRUE)
+	SHOULD_CALL_PARENT(TRUE)
+
+	is_active = TRUE
+	// Maybe add support for different stages? AKA add a stronger effect when there are more limbs
+
+/// Remove an effect whenever a threshold is no longer reached
+/datum/status_effect/bodypart_effect/proc/deactivate()
+	PROTECTED_PROC(TRUE)
+	SHOULD_CALL_PARENT(TRUE)
+
+	is_active = FALSE
+
+/// Clean up all references and self-destruct
+/datum/status_effect/bodypart_effect/Destroy()
+	deactivate()
+	for(var/obj/item/bodypart/bodypart as anything in bodyparts)
+		remove_bodypart(bodypart, deleting = TRUE)
+
+	return ..()
+
+/// This limb regens in light! Only BODYTYPE_PLANT limbs will heal, but limbs without the flag (and with the effect) still contribute to healing of the other limbs
+/datum/status_effect/bodypart_effect/photosynthesis
+	tick_interval = STATUS_EFFECT_NORMAL_PROCESS
+	id = "photosynthesis"
+
+/datum/status_effect/bodypart_effect/photosynthesis/tick(seconds_between_ticks)
+	var/light_amount = 0 //how much light there is in the place, affects receiving nutrition and healing
+	var/bodypart_coefficient = GET_BODYPART_COEFFICIENT(bodyparts)
+
+	if(isturf(owner.loc)) //else, there's considered to be no light
+		var/turf/turf_loc = owner.loc
+		light_amount = min(1, turf_loc.get_lumcount()) - 0.5
+
+		if(owner.nutrition < NUTRITION_LEVEL_ALMOST_FULL)
+			owner.adjust_nutrition(2.5 * light_amount * bodypart_coefficient)
```

Comment:
Halved the healing, nutrition etc rates and doubled the speed. Should be overal smoother

Reply:
does this mean that podpeople are no longer immune to becoming fat while in light?